### PR TITLE
added tab to navigate calendar screen

### PR DESCRIPTION
**Pull request by shreya-mishra**
Date: 19 March 2024

Developer Name: Shreya

---

## Issue Ticket Number

https://github.com/Real-Dev-Squad/mobile-app/issues/407

## Description

Tab for calendar screen

**Screeshot**
image

### Documentation Updated?

- [ ] Yes
- [✅  ] No

### Under Feature Flag

- [✅  ] Yes
- [ ] No

### Database Changes

- [ ] Yes
- [ ] No

### Breaking Changes

- [ ] Yes
- [✅  ] No

### Development Tested?

- [✅  ] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

image

</details>

## Test Coverage
<details>
<summary>Screenshot 1</summary>

image

image

</details>

### CODE REVIEW

File: __tests__/Tab.test.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    const { getByTestId } = render(
+      <Provider store={store}>
+        <TabNavigation />
+      </Provider>,
+    );
+    const calendar = getByTestId('calendar');
+
+    expect(calendar).toBeTruthy();
+
+    // Assert that the Home screen is displayed by default
```
Comment (ankushdharkar) [inline __tests__/Tab.test.tsx]: Missing assertion?

File: __tests__/Tab.test.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { render } from '@testing-library/react-native';
+import { Provider } from 'react-redux'; // Import Provider from react-redux
```
Comment (ankushdharkar) [inline __tests__/Tab.test.tsx]: What is the purpose of the comment here?

File: __tests__/Tab.test.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      </Provider>,
+    );
+    const calendar = getByTestId('calendar');
+
+    expect(calendar).toBeTruthy();
+
+    // Assert that the Home screen is displayed by default
+  });
+
+  // You can add more test cases to cover navigation and behavior of each tab
```
Comment (ankushdharkar) [inline __tests__/Tab.test.tsx]: Is this still needed?

File: src/navigations/TabNavigation/TabNavigation.tsx [hunk 4]
```diff
@@ -60,6 +63,42 @@ const TabNavigation = () => {
...
+                return (
+                  <Image
+                    style={TabViewStyle.tab_icon}
+                    source={Images.calendar}
+                  />
+                  // focused ? (
+                  //   <CalendarIcon width={60} height={60} />
+                  // ) : (
+                  //   <CalendarIcon width={30} height={30} focused />
+                  // )
```
Comment (ankushdharkar) [inline src/navigations/TabNavigation/TabNavigation.tsx]: Why the dead code?

File: src/navigations/TabNavigation/TabNavigation.tsx [hunk 5]
```diff
@@ -60,6 +63,42 @@ const TabNavigation = () => {
             },
           }}
         />
+        {isProdEnvironment && (
```
Comment (ankushdharkar) [inline src/navigations/TabNavigation/TabNavigation.tsx]: Is this check needed?

File: src/navigations/TabNavigation/TabNavigation.tsx [hunk 6]
```diff
@@ -60,6 +63,42 @@ const TabNavigation = () => {
...
+                      fontSize: Fonts.Tab_Text_Font,
+                      color: focused
+                        ? Colors.Tab_Active_Color
+                        : Colors.Tab_Inactive_Color,
+                    }}
+                  >
+                    {Strings.Tab_Calendar}
+                  </Text>
+                );
+              },
```
Comment (ankushdharkar) [inline src/navigations/TabNavigation/TabNavigation.tsx]: Can this be cleaned up for readability?

**Review by ankushdharkar — CHANGES_REQUESTED** [review]
Please review your changes once before sending the PR for review.

File: __tests__/Tab.test.tsx (hunk 1 above)
Comment (shreya-mishra) [reply __tests__/Tab.test.tsx]: removed the line

File: __tests__/Tab.test.tsx [hunk 7]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { render } from '@testing-library/react-native';
+import { Provider } from 'react-redux';
+import { store } from '../App';
+import TabNavigation from '../src/navigations/TabNavigation/TabNavigation';
```
Comment (ankushdharkar) [inline __tests__/Tab.test.tsx]: Nit: Please read on how to sort these and why that might be helpful

File: __tests__/Tab.test.tsx (hunk 2 above)
Comment (shreya-mishra) [reply __tests__/Tab.test.tsx]: removed

File: __tests__/Tab.test.tsx (hunk 3 above)
Comment (shreya-mishra) [reply __tests__/Tab.test.tsx]: removed

File: src/navigations/TabNavigation/TabNavigation.tsx (hunk 4 above)
Comment (shreya-mishra) [reply src/navigations/TabNavigation/TabNavigation.tsx]: removed

File: src/navigations/TabNavigation/TabNavigation.tsx (hunk 5 above)
Comment (shreya-mishra) [reply src/navigations/TabNavigation/TabNavigation.tsx]: removed

File: src/navigations/TabNavigation/TabNavigation.tsx (hunk 6 above)
Comment (shreya-mishra) [reply src/navigations/TabNavigation/TabNavigation.tsx]: cleaned

File: __tests__/Tab.test.tsx (hunk 7 above)
Comment (shreya-mishra) [reply __tests__/Tab.test.tsx]: read it from [doc](https://eslint.org/docs/latest/rules/sort-imports#:~:text=When%20declaring%20multiple%20imports%2C%20a,purely%20a%20matter%20of%20style)

**Says that :**

When declaring multiple imports, a sorted list of import declarations make it easier for developers to read the code and find necessary imports later. This rule is purely a matter of style.

File: __tests__/Tab.test.tsx [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    expect(getByText('Profile')).toBeTruthy();
+  });
+
+  it('navigates to Calendar screen by default', () => {
+    const { getByTestId } = render(
+      <Provider store={store}>
+        <TabNavigation />
+      </Provider>,
+    );
+    const calendar = getByTestId('calendar');
```
Comment (ankushdharkar) [inline __tests__/Tab.test.tsx]: Why not use `Tab_Calendar` key here, like rest of the app? Would that not keep the name of the string key consistent for us?

File: src/navigations/TabNavigation/TabNavigation.tsx [hunk 9]
```diff
@@ -12,16 +12,35 @@ import HomeScreenV2 from '../../screens/HomeScreen/HomeScreenV2';
 import { useSelector } from 'react-redux';
 import { AllTaskScreenStack } from '../../screens/Stacks/AllStack';
 import { scale } from '../../utils/utils';
+import CalendarInviteScreen from '../../screens/Calendar/CalendarInviteScreen';
 
 const tab = createBottomTabNavigator();
 
+const renderLabel = ({ focused }) => {
```
Comment (ankushdharkar) [inline src/navigations/TabNavigation/TabNavigation.tsx]: `renderLabel` sounds like a generic reusable function, but it seems to only be rendering label _only_ for calendar. Is this intentional?

File: src/navigations/TabNavigation/TabNavigation.tsx [hunk 10]
```diff
@@ -12,16 +12,35 @@ import HomeScreenV2 from '../../screens/HomeScreen/HomeScreenV2';
...
+        color: focused ? Colors.Tab_Active_Color : Colors.Tab_Inactive_Color,
+      }}
+    >
+      {Strings.Tab_Calendar}
+    </Text>
+  );
+};
+
+const renderImage = () => {
+  return <Image style={TabViewStyle.tab_icon} source={Images.calendar} />;
```
Comment (ankushdharkar) [inline src/navigations/TabNavigation/TabNavigation.tsx]: Is this one-line abstraction useful?

**Review by ankushdharkar — CHANGES_REQUESTED** [review]

File: src/navigations/TabNavigation/TabNavigation.tsx (hunk 9 above)
Comment (shreya-mishra) [reply src/navigations/TabNavigation/TabNavigation.tsx]: yeah, need to change for other tabs too

File: src/navigations/TabNavigation/TabNavigation.tsx (hunk 10 above)
Comment (shreya-mishra) [reply src/navigations/TabNavigation/TabNavigation.tsx]: yes ,to render icon, but this is also going to use for other tabs too

File: __tests__/Tab.test.tsx (hunk 8 above)
Comment (shreya-mishra) [reply __tests__/Tab.test.tsx]: used !

File: src/navigations/TabNavigation/TabNavigation.tsx (hunk 9 above)
Comment (shreya-mishra) [reply src/navigations/TabNavigation/TabNavigation.tsx]: changed

File: src/navigations/TabNavigation/TabNavigation.tsx (hunk 10 above)
Comment (shreya-mishra) [reply src/navigations/TabNavigation/TabNavigation.tsx]: commited

**Comment by shreya-mishra** [comment]
> Please review your changes once before sending the PR for review.

reviewed

**Review by iamitprakash — CHANGES_REQUESTED** [review]
coverage number please

**Comment by shreya-mishra** [comment]
> coverage number, please

not able to get that as all previously written test cases are failing.

**Pull request closed without merging**